contract/models/contract_template_line.py

```diff
@@ -42,18 +42,14 @@ class ContractTemplateLine(models.Model):
         readonly=False,
     )
     quantity = fields.Float(default=1.0, required=True)
-    product_uom_category_id = fields.Many2one(
-        comodel_name="uom.category",
-        related="product_id.uom_id.category_id",
-        readonly=True,
-    )
+    allowed_uom_ids = fields.Many2many("uom.uom", compute="_compute_allowed_uom_ids")
```

Comment:
Please use `Binary` field for the domain and compute it there instead of the many2many, that is less performant.